src/h5web/visualizations/heatmap/HeatmapVisContainer.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { ReactElement, useState } from 'react';
+import { range } from 'lodash-es';
+import { HDF5SimpleShape } from '../../providers/models';
+import { useDatasetValue } from '../shared/hooks';
+import { assertDataset } from '../../providers/utils';
+import DimensionMapper from '../../dataset-visualizer/mapper/DimensionMapper';
+import { DimensionMapping } from '../../dataset-visualizer/models';
+import MappedHeatmapVis from './MappedHeatmapVis';
+import { VisContainerProps } from '../shared/models';
+
+function HeatmapVisContainer(props: VisContainerProps): ReactElement {
+  const { entity } = props;
+  assertDataset(entity);
+
+  const value = useDatasetValue(entity.id);
+
+  const { dims } = entity.shape as HDF5SimpleShape;
```

Comment:
Hmm, yeah but it wouldn't hurt. We don't need the check for typechecking purposes, but it would be useful if the container were used on its own, outside of h5web. What do you think?

Reply:
We don't export containers yet so it should not need a check. On the other end, we might forget it if we export them in several weeks/months.

I would add an `assert`. I think it fits well the situation here where we know from the context that the dataset has a particular shape but still, if something goes wrong, it should hard-break.

Reply:
Agreed, it also strengthens the idea of keeping the checks and the containers completely decoupled.